runtime/caches.go

```diff
@@ -119,3 +119,23 @@ func (c *catalogCache) get(ctx context.Context, rt *Runtime, instId string) (*ca
 	c.cache[key] = service
 	return service, nil
 }
+
+type queryCache struct {
+	cache *simplelru.LRU
```

Comment:
Needs a mutex like in connectionCache – it's not concurrency safe

Reply:
ok by mistake I saw this comment - https://github.com/hashicorp/golang-lru/blob/master/lru.go#L14
I am wondering if I should use this cache directly since it already takes care of locking

Reply:
The reason the connection cache did its not concurrency management was to prevent a race condition in-between the `get` and the `add` call (where two concurrent gets for the same connection might cause two connections to be opened).

This might not apply here, in that case you can use the thread-safe one. One thing to consider is what if there's two concurrent requests for the same query? Probably it will just run both queries, which is okay for queries (but not for opening connections as above). Anyway, some considerations to get right here.

Reply:
I agree, its ok to run the same query twice for concurrent queries instead of adding locking.